src/lib.rs

```diff
@@ -422,7 +423,13 @@ where
 {
     #[inline]
     fn from(arr: [T; N]) -> Array<T, U> {
-        Self::from_core_array(arr)
+        // SAFETY: `Array` is a `repr(transparent)` newtype for `[T; N]` when it impls
+        // `ArrayOps<T, N>`.
+        unsafe {
+            let ptr = arr.as_ptr();
+            mem::forget(arr);
+            ptr::read(ptr.cast())
+        }
```

Comment:
@newpavlov double check me on this

Reply:
I think `mem::transmute` would be a better fit here.

Reply:
Aah nope, once again we run afoul of transmute and const generics:

```
error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
   --> src/lib.rs:428:18
    |
428 |         unsafe { mem::transmute(arr) }
```

Reply:
I am not sure it's safe to read pointer after you forgot an owned value. IIUC the value gets moved into the `forget` function and the old location becomes invalid. I think it will be better to write it like this:
```rust
unsafe {
    let val = mem::transmute_copy(&arr);
    mem::forget(arr);
    val
}
```

Reply:
`ManuallyDrop` is another option